kale/pipeline/mpca_trainer.py

```diff
@@ -0,0 +1,165 @@
+# =============================================================================
+# Author: Shuo Zhou, szhou20@sheffield.ac.uk
+#         Haiping Lu, h.lu@sheffield.ac.uk or hplu@ieee.org
+# =============================================================================
+
+"""Implementation of MPCA->Feature Selection->Linear SVM/LogisticRegression Pipeline
+"""
+
+import logging
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.feature_selection import f_classif
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import LinearSVC, SVC
+from sklearn.utils.validation import check_is_fitted
+
+from ..embed.mpca import MPCA
+
+param_c_grids = list(np.logspace(-4, 2, 7))
+classifiers = {"svc": [SVC, {"kernel": ["linear"], "C": param_c_grids}],
+               "linear_svc": [LinearSVC, {"C": param_c_grids}],
+               "lr": [LogisticRegression, {"C": param_c_grids}]}
+
+# k-fold cross validation used for grid search, i.e. searching for optimal value of C
+default_search_params = {'cv': 5}
+default_mpca_params = {"var_ratio": 0.97, "return_vector": True}
+
+
+class MPCATrainer(BaseEstimator, ClassifierMixin):
+
+    def __init__(self, classifier='svc', classifier_params='auto', mpca_params=None,
+                 n_features=None, search_params=None):
+        """Trainer of pipeline: MPCA->Feature selection->Classifier
+
+        Args:
+            classifier (str, optional): Classifier for training. Options: support vector machine (svc) or
+                logistic regression (lr). Defaults to 'svc'.
```

Comment:
Outdated docstrings. Now there are three options and you need to explain a bit so users know the difference from the docs rather than have to read the code.